pkg/planner/core/planbuilder.go

```diff
@@ -2327,6 +2331,13 @@ func (b *PlanBuilder) buildAnalyzeFullSamplingTask(
 		}
 	}
 
+	// Generate global indexes idxTask.
+	for _, indexInfo := range tbl.TableInfo.Indices {
```

Comment:
How about we collect this information in the getModifiedIndexesInfoForAnalyze. Because in that function we:
1. collect the normal indexes
2. skip and collect the MVIndexes
3. skip the global index

How about we also collect the global indexes in this function? Then we don't need to iterate the indexes again. And also helps us do the index check in one place.